lib/src/package_graph.dart

```diff
@@ -39,7 +39,8 @@ class PackageGraph {
   /// the packages' pubspecs are already fully-parsed.
   factory PackageGraph.fromSolveResult(
       Entrypoint entrypoint, SolveResult result) {
-    var packages = Map<String, Package>.fromIterable(result.packages,
+    var packages = Map<String, Package>.fromIterable(
+        result.packages.where((id) => id.source != null),
```

Comment:
Why is this necessary? Are we abusing notation somehow?

What does `id.source == null` imply?

Reply:
So `root` is filtered out? why?

And why do we retain `if (id.name == entrypoint.root.name) return entrypoint.root;`.

Reply:
Okay, let's move this special filtering hack into `_installInCache` in `global_packages.dart`.